### ISSUE TITLE
Add the ability to add a where clause for "one-to" side of "one-to-one" and "one-to-many"

### ISSUE THREAD

**Issue by firatoezcan** [prisma/prisma#16049]
## Problem

I'm trying to build a tool ontop of Prisma that handles access control defined via a ruleset.  
What I currently have is the ability to block certain tables from being queried as well as only returning a subset of tables that are accessible by the defined rules.

However while implementing I found that there is no `where` argument available for relations that are on the "one" side from a "one-to-many" relationship (or both sides with a "one-to-one" relation)

An example query that I would want to send is here:

```javascript
prisma.invoiceLine.findMany({
  where: {
    OR: [
      {
        Invoice: {
          is: {
            CustomerId: {
              equals: 1,
            },
          },
        },
      },
    ],
  },
  select: {
    InvoiceId: true,
    Track: {
      where: {
        InvoiceLine: {
          some: {
            Invoice: {
              is: {
                InvoiceId: {
                  in: [
                    327,
                    98,
                  ],
                },
              },
            },
          },
        },
      },
      select: {
        Name: true,
        Album: {
          select: {
            Title: true,
          },
        },
        Composer: true,
      },
    },
    UnitPrice: true,
    Quantity: true,
  },
});

```

This obviously tells me (correctly) that `where` does not exist on `Track` but without having a `where` clause there, my only other option is to fetch every record that is required for checking the rule and check this in userspace instead of having this be dealt with in the database layer.

This becomes a problem with a large amount of data as the rules can use relations and thus making me load a lot of unnecessary into memory to then filter out again. 

## Suggested solution

Add `where` to the "one" side of relations so it can be `null` depending on if the `where` matches or not

## Alternatives

There is the alternative that I would add the `where` clause higher up on the `invoiceLine`, but I want to fetch the data for the `invoiceLine` but have `Track` be null in this case. It's not a good DX that I can only provide 

## Additional context

The Prisma schema as well as the migrations for some test data are available here: https://github.com/firatoezcan/typegraphql-prisma-cms/tree/nexus/apps/api/prisma

**Comment by JoeRoddy** [comment]
Were you every able to find an alternative way to accomplish it @firatoezcan ?  Experiencing a similar issue.  Imagine I have user profiles that the user can mark as private and I want to only include them based on that toggle:

```
model User {
  id        Int      @id @default(autoincrement())
  profile   Profile? @relation(fields: [profileId], references: [id])
  profileId Int?     @unique
}

model Profile {
  isPrivate Boolean @default(true)
  email     String
  location  String

  id   Int   @id @default(autoincrement())
  user User?
}
```

`const data = await db.user.findMany({ include: { profile: { where: { isPrivate: false } } } });`
^This would be the intuitive way to do this, but I don't think this is possible currently.  I 100% agree with Firat, pretty silly to have to filter out this data manually after querying, and becomes orders of magnitude more complex if theres multiple levels of include statements.

**Comment by firatoezcan** [comment]
@JoeRoddy nope, I tried doing multiple client calls but the performance was not good enough for me so I switched to using Hasura

**Comment by tonypeng** [comment]
+1, also in this situation
